### PR TITLE
Dannym add py3 lint

### PR DESCRIPTION
**Pull request by dannymeloy**
Added Python2 and Python3 linting stages to Jenkinsfile and fixed corresponding linting errors

### CODE REVIEW

File: nmoscommon/webapi.py [hunk 1]
```diff
@@ -113,9 +113,9 @@ def __cmp(a, b):
         elif b is None:
             return -1
         elif a.priority == b.priority:
-            return cmp(a.accept_index, b.accept_index)
+            return (a.accept_index > b.accept_index) - (a.accept_index < b.accept_index)
         else:
-            return -cmp(a.priority, b.priority)
+            return -((a.accept_index > b.accept_index) - (a.accept_index < b.accept_index))
```
Comment (andrewbonney) [inline nmoscommon/webapi.py]: Is this right? It seems to have changed a comparison on priority to a comparison on accept_index.

**Comment by andrewbonney** [comment]
I should add, this should probably include a patch version bump.

File: nmoscommon/webapi.py (hunk 1 above)
Comment (dannymeloy) [reply nmoscommon/webapi.py]: Sorry my bad that’s copy and paste engineering

File: nmoscommon/__init__.py [hunk 2]
```diff
@@ -1,2 +1,2 @@
 from __future__ import absolute_import
-from .nmoscommonconfig import config
+from .nmoscommonconfig import config  # noqa F401
```
[the remark this answers is not in the document]
Comment (dannymeloy) [reply nmoscommon/__init__.py]: Is it not so people can do:
`from nmoscommon import config` as opposed to :
`from nmoscommon.nmoscommonconfig import config`

File: nmoscommon/httpserver.py [hunk 3]
```diff
@@ -17,74 +17,72 @@
...
 
-    threading.Thread.__init__(self)
+    daemon = True    # causes server to stop on main thread quitting
 
-  def run(self):
-    self.api = self.api_class(*self.api_args, **self.api_kwargs)
+    def __init__(self, api, port=0, host='0.0.0.0', api_args=None, api_kwargs=None, ssl=None):
+        self.api_args = api_args
+        self.api_kwargs = api_kwargs
+        if self.api_args is None:
```
Comment (simonrankine) [inline nmoscommon/httpserver.py]: Why not just do this in the default arg? i.e api_args=[] in the def statement?

File: nmoscommon/httpserver.py [hunk 4]
```diff
@@ -17,74 +17,72 @@
...
+            self.api_args = []
+        if self.api_kwargs is None:
```
Comment (simonrankine) [inline nmoscommon/httpserver.py]: Same against with default args

File: nmoscommon/httpserver.py (hunk 3 above)
Comment (andrewbonney) [reply nmoscommon/httpserver.py]: There's a Python gotcha here: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

**Pull request merged**